src/rc_compat.h

```diff
@@ -77,10 +82,22 @@ RC_BEGIN_C_DECLS
  #define rc_mutex_lock(mutex)
  #define rc_mutex_unlock(mutex)
 #else
- #ifdef _WIN32
-  typedef struct rc_mutex_t {
-    void* handle; /* HANDLE is defined as "void*" */
-  } rc_mutex_t;
+  #if defined(_WIN32)
+   typedef struct rc_mutex_t {
+   #if defined(WINVER) && WINVER >= 0x0600
+     /* Windows Vista and later can use a slim reader/writer (SRW) lock */
+     SRWLOCK srw_lock;
+     /* Current thread owner needs to be tracked (for recursive mutex usage) */
+     DWORD owner;
+     DWORD count;
+   #else
+     /* Pre-Vista must use a critical section */
+     CRITICAL_SECTION critical_section;
+   #endif
+   } rc_mutex_t;
+ #elif defined(GEKKO)
+  #include <ogcsys.h>
+  typedef mutex_t rc_mutex_t;
```

Comment:
The GEKKO code was written by another user. I have no way to test it.

See #312.

Reply:
Looking more deeply at the libogc docs indicates the usage is still wrong anyways: https://libogc.devkitpro.org/mutex_8h.html

Nearly all the functions take the mutex_t by value rather than by reference (only init takes it by reference). The LWP_MutexInit call also seems to be nonsense (passing NULL for the bool use_recursive seems unintended).

I could move fixes for that in a separate PR (presumingly some investigation might by warranted by RetroArch's side, RC_NO_THREADS might be getting defined hence why that is seemingly building at all).

Reply:
Seems RC_NO_THREADS is actually getting defined on RetroArch's side: https://github.com/libretro/RetroArch/blob/228426880babb529fae8f01598a6d34698ad3b18/griffin/griffin.c#L194